libs/providers/ofrep/src/lib/ofrep-provider.ts

```diff
@@ -0,0 +1,72 @@
+import { EvaluationContext, Provider, JsonValue, ResolutionDetails, TypeMismatchError } from '@openfeature/server-sdk';
+import {
+  EvaluationFlagValue,
+  handleEvaluationError,
+  OFREPApi,
+  OFREPApiEvaluationResult,
+  toResolutionDetails,
+} from '@openfeature/ofrep-core';
+
+export class OfrepProvider implements Provider {
+  private ofrepApi: OFREPApi;
+
+  readonly runsOn = 'server';
+  readonly metadata = {
+    name: OfrepProvider.name,
+  };
+
+  constructor(baseUrl: string) {
```

Comment:
Minor - knowing this is a WIP, I think we should introduce configuration options, especially targeting authentication headers. 

For go-sdk provider, I added header customization options [1]. This can be used for authentication header. 

[1] - https://github.com/open-feature/go-sdk-contrib/blob/69283f92c207f161c38886d9f9922a36aea76d05/providers/ofrep/README.md#configuration

Reply:
Yes totally, just did not do it yet :) 
I just pushed a change for this, including static headers and headerFactories.
I guess this should be a good start, feel free to have a look @Kavindu-Dodan.
In the core we also added a custom fetch implementation at least, this can now also be set.